django_fido/views.py

```diff
@@ -182,8 +181,8 @@ def get_user_id(self, user: AbstractBaseUser) -> str:
         If resident_key is True, we need to return an uuid string that does not disclose user identity
         """
         if SETTINGS.resident_key:
-            return uuid.uuid4().hex
-        return user.username
+            return uuid.uuid4().bytes
```

Comment:
This should probably be `uuid.uuid4().hex.encode()` to maintain the content of the return identifier. I am afraid that changing this to `bytes` will mean that resident keys already registered might stop working.

Reply:
I don't believe it would affect previous registrations. From what I'm seeing in the code when using resident keys user credentials are stored on the key and database. Those values are then used during authentication.

@variable could you confirm my assumption.

Reply:
Hi guys, I haven't touched this project for a while I can't remember how things get translated between server and client as in str vs bytes, the safest way is to register the key before this change and test if it still works after this change.

Reply:
Without going thru the code and understanding it again, my memory tells me that the key provides the user_handle to look up the authenticator record to validate against. So in theory, the backend is agnostic about what user_handle format is. I guess this goes inline with @MichalPham's assumption? I could be wrong tho.